tools/java/org/unicode/cldr/util/MapComparator.java

```diff
@@ -24,9 +24,17 @@
 import com.ibm.icu.util.ULocale;
 
 public class MapComparator<K> implements Comparator<K>, Freezable<MapComparator<K>> {
-    private RuleBasedCollator uca = (RuleBasedCollator) Collator.getInstance(ULocale.ROOT);
-    {
-        uca.setNumericCollation(true);
+    // initialize this once
+    private static final Collator uca = getUCA();
+
+    /**
+     * This does not change, so we can create one and freeze it.
+     * @return
+     */
+    private static Collator getUCA() {
```

Comment:
If we are sure this will be called at least ones by all users of MapComparator, this is fine. Otherwise would be good to use the Singleton thread-safe pattern (see other ticket).

Reply:
it's used exactly once, in initializing the `uca` static member. Singleton pattern would be fine here too.

Reply:
I'll check the other ticket and update this tomorrow…good point. let's start to propagate that pattern.

Reply:
CLDR-14180

Reply:
updated.